PFDR-4, PFDR-95, PFDR-96: Events; FixityCheckJob

aelkiss opened the pull request:
See commits below.

The `Bag` model will be renamed to `Package` after this PR is merged

malakai97 commented on app/controllers/v1/events_controller.rb:
```diff
@@ -0,0 +1,11 @@
+module V1
+  class EventsController < ApplicationController
+    def index
```
This implementation was a bit confusing to me. It could be cleaned up if the `.bag` scope were implemented as a filter that did nothing if not given a bag_id.  Absent that, perhaps this more clearly shows the intention:

```ruby
      events = if params[:bag_id]
        Event.bag(params[:bag_id])
      else
        Event
      end
      @events = policy_scope(events)
```

malakai97 commented on app/models/event.rb:
```diff
@@ -0,0 +1,8 @@
+  belongs_to :bag
+  belongs_to :user
+
+  scope :bag, ->(bag_id) { joins(:bag).where("bags.bag_id = :bag_id or external_id = :bag_id", bag_id: bag_id) }
```
Might be better implemented as a filter:
`scope :bag, ->(bag_id) { joins(:bag).where(...) if bag_id }`  

This abuses the fact that scopes always return at least the empty/no-op scope.

malakai97 commented on spec/controllers/v1/events_controller_spec.rb:
```diff
@@ -0,0 +1,52 @@
+        let(:key) { :event_id }
+        # for underprivileged users, should only render events where the bag
+        # belongs to the user
+        let(:factory) do
```
This is something I"d expect to see implemented as a fabricator.

aelkiss replied:
This is cribbed from https://github.com/mlibrary/chipmunk/blame/master/spec/controllers/v1/bags_controller_spec.rb#L11 - Or are you just saying that `Fabricate(:event, bag: Fabricate(:bag, user: user))` should get reduced down to something like `Fabricate(:event, bag_user: user: user)`?

malakai97 commented on spec/controllers/v1/events_controller_spec.rb:
```diff
@@ -0,0 +1,52 @@
+        let(:events) { [Fabricate(:event, bag: bag), Fabricate(:event, bag: bag)] }
+        before(:each) do
+          # ensure these exist
+          bag
```
`let!(:bag){...}` does exactly this

aelkiss commented:
I addressed all the feedback except for the proc in events_controller_spec here: https://github.com/mlibrary/chipmunk/pull/40/files/da2fd4c7eb6907a65c92544a7838d7a33045661a#diff-4e7ec40ea736ba39a615fa9a78a6287eR13

I'm going to go ahead and merge it so I can get the Bag -> Package rename in today, but if there's any further changes you'd like here or any clarification on the issue I didn't address, I'm happy to make those changes later.
